Ported env test to e2e

schebro opened the pull request:
**Description of the Pull Request (PR):**

Ported env test to E2E

**This fixes or addresses the following GitHub issues:**

- Fixes #3354 


**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)


Attn: @singularity-maintainers

obreitwi committed:
Do not forward 'SINGULARITY_'-variables

GodloveD committed:
tidy up the CHANGELOG.md

Commit:
Just a saving point...

Commit:
Merge remote-tracking branch 'upstream/master' into docker-tests-e2e

Commit:
Saving point

Commit:
Adding tests

Commit:
Exported var, & %v => %s

Commit:
Updated tests

Commit:
Merge remote-tracking branch 'upstream/master' into docker-tests-e2e

Commit:
Updated tests

Commit:
Added docker stuff

Commit:
Added the extra tests

Commit:
Uncommented tests

Commit:
Trying to make CCI happy

Commit:
Merge remote-tracking branch 'upstream/master' into docker-tests-e2e

Commit:
Updated/bug dockerPullTests

Commit:
Added godoc

Commit:
Updated tests name++

Commit:
Added --alow-unsigned

Commit:
Removed extra comment

ikaneshiro committed:
update changelog

ikaneshiro committed:
port definition build tests to e2e framework

ikaneshiro committed:
fix mistake and rename test function

Commit:
Merge remote-tracking branch 'upstream/master' into docker-tests-e2e

Commit:
Removed extra imports

ArangoGutierrez committed:
Merge branch 'master' into docker-tests-e2e

phpHavok committed:
Migrate existing instance tests to the E2E framework.

phpHavok committed:
Merge branch 'master' into 3352/e2e_instance_tests

phpHavok committed:
Minor changes I missed the first time around.

phpHavok committed:
Order structs by size to optimize layout.

gvallee committed:
Add a corner case test to get to 100% testing coverage

phpHavok committed:
Add deprecated notice in old test file.

schebro committed:
Moved functionality of remote_test.go to E2E framework to fix #3350

Commit:
Return error if image is already in the library

Commit:
Fixed warning message

Commit:
Warningf => Infof

GodloveD committed:
Merge branch 'master' into docker-tests-e2e

schebro committed:
Ported env test to e2e

schebro committed:
Ported env test to e2e

GodloveD committed:
Merge pull request #3416 from WestleyK/docker-tests-e2e

[E2E] Tests for Docker

GodloveD committed:
Merge branch 'master' into issue_2559_capability

GodloveD committed:
Merge branch 'master' into bugfix/do_not_forward_singularity_env

GodloveD committed:
Merge pull request #3468 from gvallee/issue_2559_capability

Add a corner case test to pkg/util/capabilities

GodloveD committed:
Merge branch 'master' into push-clean

gvallee committed:
Return error instead of calling os.Exit() in order to improve testing coverage

GodloveD committed:
Merge pull request #3471 from WestleyK/push-clean

Slightly improved error message when pushing a image to the library that already exist

gvallee committed:
Switch two calls to sylog.Fatalf to returning errors in order to avoid code to exit during testing

Commit:
Merge remote-tracking branch 'upstream/master' into unsigned-flag

Commit:
Added hidden flag

GodloveD committed:
Merge pull request #3432 from WestleyK/unsigned-flag

'--allow-unauthenticated' => '--allow-unsigned'

GodloveD committed:
updated tests to reflect that we are no longer passing SINGULARITY_* vars into container

GodloveD committed:
Merge branch 'master' into bugfix/do_not_forward_singularity_env

GodloveD committed:
go fmt! (that's what I get for doing a quick edit with vanilla vim)

GodloveD committed:
Merge branch 'bugfix/do_not_forward_singularity_env' of github.com:electronicvisions/singularity into bugfix/do_not_forward_singularity_env

GodloveD committed:
Merge branch 'master' into changelog_update

GodloveD committed:
Merge pull request #2728 from electronicvisions/bugfix/do_not_forward_singularity_env

Do not forward 'SINGULARITY_'-variables from user to container environment

GodloveD committed:
Merge branch 'master' into changelog_update

jscook2345 reviewed (approved):
LGTM

GodloveD committed:
Merge pull request #3436 from ikaneshiro/changelog_update

Changelog update

ikaneshiro committed:
always test for labels instead of conditionally

ikaneshiro committed:
change confusing test name

ikaneshiro committed:
add header comment for depricated tests

Commit:
Make sure Go modules are disabled when checking Go version (#3481)

Since we are running a simple program to check that the Go version
meets the minimum requirements, we should not need any modules listed in
go.mod.

This takes care of a problem in CentOS 6 where, seemingly, the git
version that ships with it does not work well with the Go version that
ships with it _when modules are enabled_ (then the go tool tries to go
to the network to donwload modules).

The rest of the code should be fine as it's using the vendor directory
(by way of passing -mod=vendor to the go tool).

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

ikaneshiro committed:
remove left over debugging statement

ArangoGutierrez committed:
Merge branch 'master' into 3352/e2e_instance_tests

ArangoGutierrez committed:
Merge pull request #3461 from phpHavok/3352/e2e_instance_tests

Port instance tests to the E2E framework

ikaneshiro committed:
generic function to create and populate temporary files

ikaneshiro committed:
Merge remote-tracking branch 'upstream/master' into e2e_definitions

WestleyK commented on e2e/env/env.go:
```diff
@@ -0,0 +1,93 @@
+var testenv testingEnv
+
+func singularityEnv(t *testing.T) {
+	// Singularity defines a path by default. See singularityware/singularity/etc/init.
```
>// Singularity defines a path by default. See singularityware/singularity/etc/init.

Is this necessary? or should the url be updated to https://github.com/sylabs/singularity/blob/master/etc/init?

jscook2345 replied:
@WestleyK Yeah, that should be updated. I put that in there before our repository migration I believe.

WestleyK commented on e2e/env/env.go:
```diff
@@ -0,0 +1,93 @@
+			b, err := cmd.CombinedOutput()
+
+			out := string(b)
+			t.Logf("args: '%v'", strings.Join(args, " "))
```
This for sure is a string, so you can use:

```suggestion
			t.Logf("args: '%s'", strings.Join(args, " "))
```

WestleyK commented on e2e/env/env.go:
```diff
@@ -0,0 +1,93 @@
+
+			out := string(b)
+			t.Logf("args: '%v'", strings.Join(args, " "))
+			t.Logf("env: '%v'", strings.Join(cmd.Env, " "))
```
Here to:

```suggestion
			t.Logf("env: '%s'", strings.Join(cmd.Env, " "))
```

WestleyK reviewed (commented):
Couple suggestions...

ikaneshiro committed:
fix incorrect ImageVerify calls

ikaneshiro committed:
Merge branch 'master' into remote_e2e

Commit:
Pass -mod=vendor to go list to avoid downloading modules (#3482)

Since we have vendored copies of the Go modules' code, we need to pass
-mod=vendor almost everywhere where we call go to prevent go from trying
to download them to the module cache.

makeit/gengodep is missing that.

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

schebro committed:
Resolve merge conflicts to fix #3470

ikaneshiro committed:
Merge branch 'master' into remote_e2e

ikaneshiro committed:
Merge pull request #3470 from schebro/remote_e2e

Moved functionality of remote_test.go to E2E framework to fix #3350

ikaneshiro reviewed (approved):
lgtm once these minor changes are addressed

ikaneshiro committed:
Merge branch 'master' into issue_3315_exit_2

ikaneshiro committed:
Merge pull request #3475 from gvallee/issue_3315_exit_2

Return error instead of explicitly failing

ArangoGutierrez committed:
Merge branch 'master' into e2e_definitions

jmstover committed:
RPM Spec fixes (#3462)

* RPM Spec fixes

Was having build issues on SLES 11. This cleans up the `%files`
listing, and puts in some extra logic if you're on SLES 11.

* Remove the glob include of singularity config directory

Commit:
Add missing  oci state commang back in CLI

cclerget committed:
Merge pull request #3483 from sashayakovtseva/fix-state-cmd

Add missing  oci state commang back in CLI

Commit:
Updated cache clean docs (#3469)

jscook2345 committed:
Merge branch 'master' into e2e_definitions

ikaneshiro committed:
Merge pull request #3435 from ikaneshiro/e2e_definitions

Migrate Definition Build Tests to E2E Framework

schebro committed:
Port help test to e2e to fix #3353

ikaneshiro committed:
remove libexec package

ikaneshiro committed:
Merge pull request #3485 from ikaneshiro/remove_libexec

Remove Libexec Package

ArangoGutierrez committed:
Merge pull request #3489 from schebro/help_e2e

Port help test to e2e

schebro committed:
Merge branch 'env_e2e' of https://github.com/schebro/singularity into env_e2e
